lib/shim/shim.js

```diff
@@ -1326,7 +1326,9 @@ function getSegment(obj) {
  */
 function getActiveSegment(obj) {
   var segment = this.getSegment(obj)
-  if (segment && segment.transaction && segment.transaction.isActive()) {
+  if (segment && segment.transaction &&
```

Comment:
You should else this with some instrumentation to send a stack trace and other diagnostics to NewRelic so that all of the instances of this problem can be aggregated and give NewRelic devs more information about how to fix this and the code paths involved.  That way though the data does get dropped in these cases, we learn something each time it happens and it can be quantified.